server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java

```diff
@@ -636,7 +638,21 @@ protected BasicDataSource getDatasource()
   {
     MetadataStorageConnectorConfig connectorConfig = getConfig();
 
-    BasicDataSource dataSource = new BasicDataSource();
+    BasicDataSource dataSource = null;
+
+    try {
+      Properties dbcpProperties = connectorConfig.getProperties();
```

Comment:
Presently, users can give a [PasswordProvider](http://druid.io/docs/latest/operations/password-provider.html) object when they specify the `druid.metadata.storage.connector.password` property. It would be helpful if the users have the flexibility to provide the `PasswordProvider` object within the dbcp config file instead of the plaintext password.

Reply:
The point of adding this change was not to change existing behavior for properties which are directly settable in existing config.  The addition was to simply address the remaining properties which are not directly settable.  I can add to doc that the username, password, connect uri, validation query, and test on borrow flag must be set outside the dbcp properties file.

Reply:
Thanks for clarifying this in the docs.